### ISSUE TITLE
Improve performance when providing an instance

### ISSUE THREAD

**Issue by facundobatista** [canonical/craft-providers#91]
Currently it's always reinjecting snaps regardless if it needs to or not.

It also always runs apt update, it does not need to do that if no changes have happened, nor does it need to install the same dependency over and over.

**Comment by mr-cal** [comment]
@facundobatista, what do you mean by this?
> if no changes have happened

Is this what you would expect when developing a charm?
```
> charmcraft pack
# first time running, so apt update and snap injection occurs

> charmcraft pack
# no apt update nor snap injection occurs

> vim charmcraft.yaml
# make a config change

> charmcraft pack
# project changed, so apt update and snap injection occurs
```

**Comment by sergiusens** [comment]
Using your flow,

```
> charmcraft pack
# first time running, so apt update and snap injection occurs

> charmcraft pack
# no apt update nor snap injection occurs

> vim charmcraft.yaml
# make a config change

> charmcraft pack
# no apt update nor snap injection occurs
# craft-parts handles cache updates for changes in stage-packages and build-packages
```

**Comment by sergiusens** [comment]
Fixed in ##110
